src/Http.hs

```diff
@@ -0,0 +1,21 @@
+{-# LANGUAGE OverloadedStrings #-}
+module Http (start) where
+
+import           Data.String
+import           Control.Monad
+import           Control.Concurrent
+import           Network.Wai
+import           Network.HTTP.Types
+import           Network.Wai.Handler.Warp (run)
+
+start :: IO String -> IO ()
+start trigger = do
+  putStrLn $ "http://localhost:" ++ show port
```

Comment:
Yes, I agree that it's more correct, but practically I don't think it's significant.  The reason I prefer is this way is that that currently the output happens in the main thread.  If we put it into a `beforeMainLoop`, it will be a different thread, and multiple threads will mess with the standard streams, so we would need synchronization.